QueryManager/src/main/java/eu/neclab/ngsildbroker/queryhandler/controller/QueryController.java

```diff
@@ -83,32 +80,53 @@ public Uni<RestResponse<Object>> getEntity(HttpServerRequest request, @QueryPara
 			@QueryParam(value = "localOnly") boolean localOnly, @PathParam("entityId") String entityId,
 			@QueryParam(value = "doNotCompact") boolean doNotCompact,
 			@QueryParam("containedBy") @DefaultValue("") String containedBy, @QueryParam("join") String join,
-			@QueryParam("idsOnly") boolean idsOnly, @QueryParam("joinLevel") @DefaultValue("1") int joinLevel) {
+			@QueryParam("idsOnly") boolean idsOnly, @QueryParam("joinLevel") @DefaultValue("1") int joinLevel,
+			@QueryParam("pick") String pick,@QueryParam("omit") String omit) {
 		int acceptHeader = HttpUtils.parseAcceptHeader(request.headers().getAll("Accept"));
 		if (acceptHeader == -1) {
 			return HttpUtils.getInvalidHeader();
 		}
-
+		if (omit!= null && pick != null) {
+			return Uni.createFrom()
+					.item(HttpUtils.handleControllerExceptions(new ResponseException(ErrorType.BadRequestData)));
+		}
 		List<Object> headerContext;
 		headerContext = HttpUtils.getAtContext(request);
 		logger.debug("retrieve called: " + request.path());
 		return HttpUtils.getContext(headerContext, ldService).onItem().transformToUni(context -> {
 //			AttrsQueryTerm attrsQuery;
 			LanguageQueryTerm langQuery;
-			try {
+			String finalPick;
+			List<String> omitList = omit == null ? new ArrayList<>() : Arrays.asList(omit.replaceAll("[\"\\n\\s]", "").split(","));
```

Comment:
No need to this logic if you take omit and pick parameter as list of string.

Reply:
done